### PR TITLE
Pos portal

### PR DESCRIPTION
**Pull request by arthcp**
Added class for POS Portal RootChainManager contract.
Exposed these methods in index.ts.
Added relevant examples and updated README.

### CODE REVIEW

File: src/root/POSRootChainManager.ts [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+    const _options = await this._fillOptions(options, txObject, this.web3Client.getParentWeb3())
+
+    if (options.encodeAbi) {
+      _options.data = txObject.encodeABI()
+      _options.to = rootToken
+      return _options
+    }
+
+    return this.web3Client.send(txObject, _options)
```
Comment (atvanguard) [inline src/root/POSRootChainManager.ts:49]: This chunk is duplicated too many times, can we put this in a common function? - This is also the case for methods in default Matic client, could file an issue to fix that later.

File: src/root/POSRootChainManager.ts [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        ethUtils.bufferToHex(receiptsRoot),
+        ethUtils.bufferToHex(receipt),
+        ethUtils.bufferToHex(ethUtils.rlp.encode(receiptParentNodes)),
+        ethUtils.bufferToHex(ethUtils.rlp.encode(path)),
+        logIndex,
+        burnTxHash,
+      ])
+    )
+  }
+}
```
Comment (atvanguard) [inline src/root/POSRootChainManager.ts:191]: This chunk is duplicated (except an extra burnTxHash field) WithdrawManager.ts. Placing them in common might help.

Comment (arthcp) [reply src/root/POSRootChainManager.ts]: Moved this to `ExitManager` class.

File: src/root/POSRootChainManager.ts (hunk 1 above)
Comment (arthcp) [reply src/root/POSRootChainManager.ts:42]: Filing an issue for this.

File: src/root/WithdrawManager.ts [hunk 3]
```diff
@@ -15,65 +14,46 @@ import Web3Client from '../common/Web3Client'
...
+  constructor(withdrawManager: address, rootChain: RootChain, web3Client: Web3Client, registry: Registry) {
     super(web3Client)
-    this.withdrawManager = new this.web3Client.parentWeb3.eth.Contract(
-      WithdrawManagerArtifact.abi,
-      withdrawManager,
-    )
+    this.withdrawManager = new this.web3Client.parentWeb3.eth.Contract(WithdrawManagerArtifact.abi, withdrawManager)
     this.rootChain = rootChain
     this.registry = registry
+    this.exitManager = new ExitManager(rootChain, web3Client)
```
Comment (atvanguard) [inline src/root/WithdrawManager.ts:40]: WithdrawManager should inherit from ExitManager. I'll take a look later. Thanks for the fix.

**Comment by ptsayli** [comment]
We want dist folder to support CDN
I have made changes in config, now lib is build directory that is ignored
@arthcp Please run `npm run build:legacy` and push files

**Comment by arthcp** [comment]
Pushed dist file @ptsayli @atvanguard

**Pull request merged**